daemon/algod/api/server/v2/handlers.go

```diff
@@ -1114,6 +1115,70 @@ func (v2 *Handlers) GetApplicationByID(ctx echo.Context, applicationID uint64) e
 	return ctx.JSON(http.StatusOK, response)
 }
 
+// GetApplicationBoxes returns the box names of an application
+// (GET /v2/applications/{application-id}/boxes)
+func (v2 *Handlers) GetApplicationBoxes(ctx echo.Context, applicationID uint64, params generated.GetApplicationBoxesParams) error {
+	appIdx := basics.AppIndex(applicationID)
+	ledger := v2.Node.LedgerForAPI()
+	lastRound := ledger.Latest()
+	keyPrefix := logic.MakeBoxKey(appIdx, "")
+
+	maxBoxThreshold := v2.Node.Config().MaxAPIBoxPerApplication
+
+	isNilOrZero := func(a *uint64) bool { return a == nil || *a == 0 }
```

Comment:
if we just check and cast the `params.Max` here, we could remove null checks entirely and make comparisons easier (e.g. `dominatedByQryParams := maxBoxThreshold > *params.Max || maxBoxThreshold == 0`):

```
if params.Max == nil {
    *params.Max = 0 // or some other var
}
```

Reply:
casting first would be good, but we can't get away with the check over `*params.Max` in `dominatedByQryParams` and `returnAll`:
- `dominatedByQryParams`: in this case, we want `*params.Max` to be lower than algod config, so we should ensure `*params.Max` to be non-zero.
- `isReturnAll`: in this case, we want both algod config and param to be zero/nil.

I can lift the cast ahead, but let me know if there is space to simplify the logical condition.